File: cpp-lab/exp-7.cpp
#include <iostream>

class Bank {
  private:
    int accNum;
    std::string name, accType;
    double balance;
  public:
    void setProperties (int a, std::string t, std::string n, double b) {
      accNum = a;
      name = n;
      accType = t;
      balance = b;
    }
    void deposit (double amount) {
      balance += amount;
    }
    void withdraw (double amount) {
      if (balance - amount < 0) std::cout << "Insufficient balance" << std::endl;
      else balance -= amount;
    }
    void display () {
      std::cout << "Name: " << name << std::endl;
      std::cout << "Balance: " << balance << std::endl;
    }
};

int main () {
  Bank bnk;
  int accNum, choice;
  std::string name, accType;
  double amount;

  do {
    std::cout << "\nWelcome to Bank\n 1. Set Bank Information\n 2. Display balance\n 3. Deposit\n 4. Withdraw\n 5. Exit\n Enter your choice (1/2/3/4): ";
    std::cin >> choice;
    switch (choice) {
      case 1:
        std::cout << "\nEnter bank account number, type, your name and balance: ";
        std::cin >> accNum >> accType >> name >> amount;
        bnk.setProperties(accNum, accType, name, amount);
        break;
      case 2:
        bnk.display();
        break;
      case 3:
        std::cout << "\nEnter the amount to deposit: ";
        std::cin >> amount;
        bnk.deposit(amount);
        break;
      case 4:
        std::cout << "\nEnter the amount to withdraw: ";
        std::cin >> amount;
        bnk.withdraw(amount);
        break;
    }
  } while (choice < 5);

  std::cout << "\nExiting...\n";

  return 0;
}

File: sem4/java-lab/Exp7.java
// Q1.  Objective:
//     • Use the Scanner class to take user input.
//     • Implement a simple bank account system with deposit and withdrawal
//     features.
// A bank wants a simple program where customers can:
// ✅ Check balance
// ✅ Deposit money
// ✅ Withdraw money
// Create a class having functions to accept multiple customer data and
// performing above given operations.

import java.util.Scanner;

class Exp7 {

  int balance = 0;
  String name, accType;

  void checkBalance() { System.out.println("Balance: " + balance); }

  void deposit(int amount) {
    balance += amount;
    System.out.println("Deposited: " + amount);
  }

  void withdraw(int amount) {
    if (balance < amount) {
      System.out.println("Insufficient balance");
    } else {
      balance -= amount;
      System.out.println("Withdrawn: " + amount);
    }
  }

  void display() {
    System.out.println("Name: " + name);
    System.out.println("Account Type: " + accType);
    System.out.println("Balance: " + balance);
  }

  public static void main(String args[]) {

    Scanner sc = new Scanner(System.in);

    int choice, amount;
    String name, accType;

    int n;

    System.out.print("Enter number of customers: ");
    n = sc.nextInt();

    Exp7 obj[] = new Exp7[n];

    for (int i = 0; i < n; i++) {
      obj[i] = new Exp7();
      System.out.print("Enter name: ");
      obj[i].name = sc.next();
      System.out.print("Enter account type: ");
      obj[i].accType = sc.next();
    }

    do {
      System.out.println("1. Check balance");
      System.out.println("2. Deposit money");
      System.out.println("3. Withdraw money");
      System.out.println("4. Display details");
      System.out.println("5. Exit");
      System.out.print("Enter your choice: ");
      choice = sc.nextInt();

      switch (choice) {
      case 1:
        System.out.print("Enter account number (1 - " + n + "): ");
        int accNo = sc.nextInt();
        obj[accNo - 1].checkBalance();
        break;
      case 2:
        System.out.print("Enter account number (1 - " + n + "): ");
        accNo = sc.nextInt();
        System.out.print("Enter amount to deposit: ");
        amount = sc.nextInt();
        obj[accNo - 1].deposit(amount);
        break;
      case 3:
        System.out.print("Enter account number (1 - " + n + "): ");
        accNo = sc.nextInt();
        System.out.print("Enter amount to withdraw: ");
        amount = sc.nextInt();
        obj[accNo - 1].withdraw(amount);
        break;
      case 4:
        System.out.print("Enter account number (1 - " + n + "): ");
        accNo = sc.nextInt();
        obj[accNo - 1].display();
        break;
      case 5:
        break;
      default:
        System.out.println("Invalid choice");
      }
    } while (choice != 5);
  }
}
